table/two_level_iterator.cc

```diff
@@ -43,6 +43,10 @@ class TwoLevelIndexIterator : public InternalIteratorBase<IndexValue> {
     assert(Valid());
     return second_level_iter_.key();
   }
+  Slice user_key() const override {
```

Comment:
This fixes a corruption bug. If we inherit the base class's `user_key()` implementation, it will `ExtractUserKey()` on `key()`. `key()` simply returns whatever is the current key in the index block, which may be an internal key or a user key.  In case it's a user key, calling `ExtractUserKey()` on it causes an incorrect truncation.